### PR TITLE
Add tests for tessellation pipeline creation VUs

### PR DESCRIPTION
**Pull request by cdwfs**

### CODE REVIEW

**Comment by cdwfs** [comment]
This may have some conflicts with #1710; I'm happy to sit on this PR until that one's merged, and handle the fixups myself.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2860,15 +2860,15 @@ static bool verifyPipelineCreateState(layer_data *dev_data, std::vector<PIPELINE
...
     }
     // Either both or neither TC/TE shaders should be defined
-    if ((pPipeline->active_shaders & VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT) &&
-        !(pPipeline->active_shaders & VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT)) {
+    bool has_control = (pPipeline->active_shaders & VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT);
+    bool has_eval = (pPipeline->active_shaders & VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT);
+    if (has_control && !has_eval) {
         skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
                         reinterpret_cast<uint64_t &>(pPipeline->pipeline), __LINE__, VALIDATION_ERROR_00534, "DS",
                         "Invalid Pipeline CreateInfo State: TE and TC shaders must be included or excluded as a pair. %s",
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Previously you'd always get /an/ error if the pipeline was invalid, but this is closer to what the VUs actually say. :+1: 

Incidentally, it's awful that we have the `has_control ^ has_eval` rule split out into two VUs with different IDs but the same message...

File: layers/parameter_validation.cpp [hunk 2]
```diff
@@ -2958,10 +2958,10 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateGraphicsPipelines(VkDevice device, VkPipeli
 
                 if (pCreateInfos[i].pTessellationState->sType != VK_STRUCTURE_TYPE_PIPELINE_TESSELLATION_STATE_CREATE_INFO) {
                     skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0,
-                                    __LINE__, VALIDATION_ERROR_00538, LayerName,
+                                    __LINE__, VALIDATION_ERROR_01427, LayerName,
                                     "vkCreateGraphicsPipelines: parameter pCreateInfos[%d].pTessellationState->sType must be "
```
Comment (chrisforbes) [inline layers/parameter_validation.cpp]: :+1:

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -15030,6 +15030,140 @@ TEST_F(VkLayerTest, CreatePipelineTessPatchDecorationMismatch) {
...
+        pipe.AddShader(&fs);
+
+        // Pass a tess control shader without a tess eval shader
+        pipe.AddShader(&tcs);
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, VALIDATION_ERROR_00534);
+        pipe.CreateVKPipeline(descriptorSet.GetPipelineLayout(), renderPass());
+        m_errorMonitor->VerifyFound();
+    }
+
+    {
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: New tests look clean :+1:

**Comment by cdwfs** [comment]
AppVeyor failed due to an implicit int -> bool promotion warning ([C4800](https://msdn.microsoft.com/en-us/library/b6801kcy.aspx)) being promoted to an error. The fix was easy, but neither the buildbots nor the Visual Studio project generated by CMake treat this warning as an error; any way we could get them all to agree on the same list of fatal warnings?

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -15030,6 +15030,140 @@ TEST_F(VkLayerTest, CreatePipelineTessPatchDecorationMismatch) {
...
+        VkPipelineTessellationStateCreateInfo tsci_bad = tsci;
+        tsci_bad.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+        pipe.SetTessellation(&tsci_bad);
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, VALIDATION_ERROR_01427);
+        pipe.CreateVKPipeline(descriptorSet.GetPipelineLayout(), renderPass());
+        m_errorMonitor->VerifyFound();
+        // Pass out-of-range patchControlPoints
+        tsci_bad = tsci;
+        tsci_bad.patchControlPoints = 0;
+        pipe.SetTessellation(&tsci);
```
Comment (mikes-lunarg) [inline tests/layer_validation_tests.cpp]: This SetTessellation call seems like it will just be overridden by the one below

**Pull request merged**